usecases/objects/manager.go

```diff
@@ -127,8 +129,12 @@ type ModulesProvider interface {
 func NewManager(locks locks, schemaManager schemaManager,
 	config *config.WeaviateConfig, logger logrus.FieldLogger,
 	authorizer authorizer, vectorRepo VectorRepo,
-	modulesProvider ModulesProvider, metrics objectsMetrics,
+	modulesProvider ModulesProvider, metrics objectsMetrics, allocChecker *memwatch.Monitor,
 ) *Manager {
+	if allocChecker == nil {
+		allocChecker = memwatch.NewDummyMonitor()
+	}
```

Comment:
Can we remove this part? This looks to me like a fix for tests, where we forgot to pass the `allocChecker`. Can we always explicitly create `allocChecker` and remove this part of code?

Reply:
I would rather like to keep it. We have to many null-pointer panics and without something like this we need to check for nil at every call-site which is easily forgotten, eg like here https://github.com/weaviate/weaviate/pull/4643#discussion_r1559418029